server/accessors/appointmentAccessor.class.php

```diff
@@ -11,13 +11,65 @@ class AppointmentAccessor {
 	public function cancelAppointment($appointmentId) {
 		GLOBAL $DB_CONN;
 
-		$query = "INSERT INTO ServicedAppointment (appointmentId, notCompletedDescription, completed)
-			VALUES (?, 'Cancelled Appointment', FALSE)";
-		$stmt = $DB_CONN->prepare($query);
+		$query = 'INSERT INTO ServicedAppointment (servicedAppointmentId, appointmentId, cancelled, completed)
+			VALUES (
+				(SELECT sa2.servicedAppointmentId FROM ServicedAppointment sa2 WHERE appointmentId = ?),
```

Comment:
So in this new query, we don't assume that a ServicedAppointment has not been created for the appointment. We use this `INSERT INTO _ ON DUPLICATE KEY UPDATE` clause that MySQL supports which inserts the entry if there exists no entry with the PK, and if the PK exists, it updates the entry based on what comes after the UPDATE portion of the clause. So the inner query here finds the PK of the entry if one exists, otherwise it returns NULL, and conveniently, if you pass NULL as the PK in MySQL, it just does the typical auto_increment functionality. Kinda neat.

For reference: https://dev.mysql.com/doc/refman/5.5/en/insert-on-duplicate.html